app/controllers/api/providers_controller.rb

```diff
@@ -104,9 +104,10 @@ def change_password_resource(type, id, data = {})
       end
     end
 
-    def verify_credentials_resource(_type, _id, data = {})
+    def verify_credentials_resource(_type, id = nil, data = {})
       klass = fetch_provider_klass(collection_class(:providers), data)
       zone_name = data.delete('zone_name')
+      data['id'] = id if id
```

Comment:
should this be `data['id'] ||= id if id` to not overwrite a pre-existing value?

Reply:
@chessbyte not really, the request comes in with a `data['id']` but that value is deleted and moved into the `id`. So I'm actually moving it back to the `data` for future re-use. There's no way that `data['id']` is set and even if that would be true, its value would always equal to `id`.